### PR TITLE
Assignment4 - Pull Request - Kris Veselits

### PR DESCRIPTION
**Pull request by kveselits**

### CODE REVIEW

File: Assignment4/UniversityCourse.Tests/CourseTests.cs [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+            bool containsStudent = false;
+            foreach (Person student in testCourse.StudentRoster)
+            {
+                if (student.Name.Equals("Dave White"))
+                {
+                    containsStudent = true;
+                }
+            }
+            Assert.IsTrue(containsStudent);
+            //There's probably a better way to test this.
```
Comment (valdynhunt) [inline Assignment4/UniversityCourse.Tests/CourseTests.cs]: One possible addition might be to separate into two tests, where the one that you have verifies the name added, and another  test, maybe called Add_Student_Increments_Class_Roster would deal with the size of the roster before and after the add.

File: Assignment4/UniversityCourse/Course.cs [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            ProfessorName = professorName;
+            ScheduleItem = (Event)this;
+            CourseSchedule = new List<Event> { ScheduleItem };
+            StudentRoster = new List<Person>();
+            CourseInstantiationCount++;
+            UniversityName = "Eastern Washington University";
+        }
+
+        public string UniversityName { get; }
+
```
Comment (valdynhunt) [inline Assignment4/UniversityCourse/Course.cs]: You may want to add validation as well as possibly throwing an Exception when an Argument is invalid or null.

File: Assignment4/UniversityCourse/Application.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+namespace UniversityCourse
+{
+    public static class Application
+    {
+        public static string Display(Object eventItem)
+        {
+            switch (eventItem)
+            {
+                case Course universityCourse:
```
Comment (MarkMichaelis) [inline Assignment4/UniversityCourse/Application.cs]: See See [Assignment4/GradingNotes.md #6](https://github.com/IntelliTect-Samples/EWU-CSCD371-2018-Fall/blob/master/Assignment4/GradingNotes.md#when-using-pattern-matching-with-a-polymorphism-ie-universitycourse-derives-from-event-and-either-or-both-have-a-getsummaryinformation-method-should-i-have-case-statments-for-both-types-ie--universitycourse-and-event).

File: Assignment4/UniversityCourse/Course.cs [hunk 4]
```diff
@@ -0,0 +1,67 @@
+using System;
+using System.Collections.Generic;
+
+namespace UniversityCourse
+{
+    public class Course : Event
+    {
+        private static int _CourseInstantiationCount;
+        private List<Event> _CourseSchedule;
```
Comment (MarkMichaelis) [inline Assignment4/UniversityCourse/Course.cs]: It is good to see your use of generics... nice!

File: Assignment4/UniversityCourse/Course.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        private List<Person> _StudentRoster;
+        private Professor _ProfessorName;
+
+        public Course(string name, string place, Professor professorName, DateTime startTime, DateTime endTime)
+            : base(name, place, startTime, endTime)
+        {
+            ProfessorName = professorName;
+            ScheduleItem = (Event)this;
+            CourseSchedule = new List<Event> { ScheduleItem };
```
Comment (MarkMichaelis) [inline Assignment4/UniversityCourse/Course.cs]: You are already storing `this` in `ScheduleItem` and now again in `CourseSchedule`.  When would one use `ScheduleItem` (in place of CourseSchedule)?  The OO design here is confusing.

File: Assignment4/UniversityCourse/Course.cs [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+            CourseInstantiationCount++;
+            UniversityName = "Eastern Washington University";
+        }
+
+        public string UniversityName { get; }
+
+        public Professor ProfessorName { get => _ProfessorName; set => _ProfessorName = value; }
+        public List<Event> CourseSchedule { get => _CourseSchedule; set => _CourseSchedule = value; }
+        public List<Person> StudentRoster { get => _StudentRoster; set => _StudentRoster = value; }
+        public int CourseInstantiationCount { get => _CourseInstantiationCount; set => _CourseInstantiationCount = value; }
```
Comment (MarkMichaelis) [inline Assignment4/UniversityCourse/Course.cs]: See [Assignment4/GradingNotes.md #19](https://github.com/IntelliTect-Samples/EWU-CSCD371-2018-Fall/blob/master/Assignment4/GradingNotes.md#19-should-i-keep-track-of-universitycourse-instances-separately-from-event-instance-count).

**Pull request closed without merging**